velox/functions/prestosql/Slice.cpp

```diff
@@ -200,22 +203,35 @@ class SliceFunction : public exec::VectorFunction {
     return std::min(endIndex - start, length);
   }
 };
+} // namespace
+
+template <TypeKind Kind>
+void registerSliceFunction(const std::string& prefix) {
+  using NativeType = typename TypeTraits<Kind>::NativeType;
+  auto kindName = TypeTraits<Kind>::name;
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+  if constexpr (
+      std::is_same_v<NativeType, int64_t> ||
+      std::is_same_v<NativeType, int32_t>) {
+    signatures.push_back(exec::FunctionSignatureBuilder()
+                             .typeVariable("T")
+                             .returnType("array(T)")
+                             .argumentType("array(T)")
+                             .argumentType(kindName)
+                             .argumentType(kindName)
+                             .build());
+  } else {
+    VELOX_FAIL(
+        "Unsupported parameter type {} to register slice function", kindName);
+  }
 
-static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
-  return {// array(T, bigint, bigint) -> array(T)
-          exec::FunctionSignatureBuilder()
-              .typeVariable("T")
-              .returnType("array(T)")
-              .argumentType("array(T)")
-              .argumentType("bigint")
-              .argumentType("bigint")
-              .build()};
+  exec::registerVectorFunction(
+      prefix + "slice", signatures, std::make_unique<SliceFunction<Kind>>());
 }
 
-} // namespace
+template void registerSliceFunction<TypeKind::BIGINT>(
+    const std::string& prefix);
+template void registerSliceFunction<TypeKind::INTEGER>(
```

Comment:
I note the above registering code is only required by Spark. Also considering this function's implementation is shared by Presto & Spark, I think we can move this function into a common place from prestosql folder, e.g., `functions/lib/`. @mbasmanova, do you agree?

Reply:
Thanks @PHILO-HE @mbasmanova let me try to fix it.